src/test/java/org/folio/rest/impl/PasswordResourceTest.java

```diff
@@ -191,6 +194,91 @@ public void shouldReturnSuccessfulValidationWhenPasswordPassesAllRules(final Tes
       .body(PASSWORD_VALIDATION_RESULT_JSON_PATH, is(ValidatorHelper.VALIDATION_VALID_RESULT));
   }
 
+  @Test
+  public void shouldReturnFailedValidationResultWhenUserNotFound() {
+    ResponseDefinitionBuilder mockDefinition = WireMock.okJson(new JsonObject()
+      .put(USERS_KEY, new JsonArray())
+      .put(TOTAL_RECORDS_KEY, 0).toString());
+    initMockUserService(mockDefinition);
+
+    Password passwordToValidate = new Password().withPassword("P@sword12");
+    RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .header(TENANT_HEADER)
+      .header(USER_ID_HEADER)
+      .header(TOKEN_HEADER)
+      .header(userMockUrlHeader)
+      .body(passwordToValidate)
+      .when()
+      .post(VALIDATE_PATH)
+      .then()
+      .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void shouldReturnFailedValidationResultWhenUserApiReturnBadRequest() {
+    ResponseDefinitionBuilder mockDefinition = WireMock.badRequest();
+    initMockUserService(mockDefinition);
+
+    Password passwordToValidate = new Password().withPassword("P@sword12");
+    RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .header(TENANT_HEADER)
+      .header(USER_ID_HEADER)
+      .header(TOKEN_HEADER)
+      .header(userMockUrlHeader)
+      .body(passwordToValidate)
+      .when()
+      .post(VALIDATE_PATH)
+      .then()
+      .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void shouldReturnFailedValidationResultWhenUserReturnIncorrectRequest() {
+    ResponseDefinitionBuilder mockDefinition = WireMock.okJson(new JsonObject()
+      .put(TOTAL_RECORDS_KEY, 0).toString());
+    initMockUserService(mockDefinition);
+
+    Password passwordToValidate = new Password().withPassword("P@sword12");
+    RestAssured.given()
+      .port(port)
+      .contentType(MediaType.APPLICATION_JSON)
+      .header(TENANT_HEADER)
+      .header(USER_ID_HEADER)
+      .header(TOKEN_HEADER)
```

Comment:
port, content type and headers can be moved to request specification as them used several times

Reply:
done